modules/uhttp/middleware.go

```diff
@@ -35,91 +35,67 @@ import (
 
 const _panicResponse = "Server Error"
 
-// InboundMiddleware applies inbound middleware on requests or responses such as
-// adding tracing to the context.
-type InboundMiddleware interface {
-	Handle(w http.ResponseWriter, r *http.Request, next http.Handler)
-}
-
-// InboundMiddlewareFunc is an adaptor to call normal functions to apply inbound middleware.
-type InboundMiddlewareFunc func(w http.ResponseWriter, r *http.Request, next http.Handler)
-
-// Handle implements Handle from the InboundMiddleware interface and simply delegates to the function
-func (f InboundMiddlewareFunc) Handle(w http.ResponseWriter, r *http.Request, next http.Handler) {
-	f(w, r, next)
-}
-
-type contextInbound struct {
-	log *zap.Logger
-}
-
-func (f contextInbound) Handle(w http.ResponseWriter, r *http.Request, next http.Handler) {
-	next.ServeHTTP(w, r.WithContext(r.Context()))
+func contextInbound(next http.Handler, log *zap.Logger) http.HandlerFunc {
+	return func(w http.ResponseWriter, r *http.Request) {
+		next.ServeHTTP(w, r.WithContext(r.Context()))
+	}
 }
 
-type tracingInbound struct{}
+func tracingInbound(next http.Handler) http.HandlerFunc {
+	return func(w http.ResponseWriter, r *http.Request) {
+		ctx := r.Context()
+		operationName := r.Method
+		carrier := opentracing.HTTPHeadersCarrier(r.Header)
+		spanCtx, err := opentracing.GlobalTracer().Extract(opentracing.HTTPHeaders, carrier)
```

Comment:
I don't think we need a global tracer, but can get it as a constructor parameter.
Anyway, it can be addressed in following PRs.